src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/PropertyMetadata.cs

```diff
@@ -62,7 +62,10 @@ public PropertyMetadata(IProperty property, Type dbContextType)
 
             IsEnum = property.ClrType.GetTypeInfo().IsEnum;
             IsReadOnly = property.IsReadOnlyBeforeSave;
-            IsAutoGenerated = property.IsStoreGeneratedAlways;
+
+            IsAutoGenerated = ((property.RequiresValueGenerator
+                || property.IsStoreGeneratedAlways)
+                && (property.ValueGenerated != ValueGenerated.Never));
```

Comment:
@prafullbhosale It depends... In EF Core, the general idea is that if an application specifies a value (i.e. sets the property to something other than the CLR default for the type), then that value will be pushed to the database regardless of whether the property is configured to use store generated values. A common use for this is when people want to insert specific seed data into the database for testing, but use generated values on normal running of the application. So at the highest level, any property _can_ be set by the application. 

If we look at this from a different perspective and ask, "can EF generate a value for this property?" then the answer is yes if:
- The property is marked as ValueGenerated.OnAdd or ValueGenerated.OnAddOrUpdate
- Or the property has a ValueGenerator explicitly set

It seems to me that using this is a reasonable heuristic for scaffolding, but it could result in situations where people want to set the value (e.g. for seeding/testing) but can't.

Three wrinkles to keep in mind with this:
- Values for foreign key properties will be populated by EF if a graph of entities including the entity where the PK is defined is being inserted together. If a single entity is being inserted, then FK values will likely need to be set by the application.
- Sometimes database columns are set up in such a way that explicit values cannot be inserted. For example, SQL Server Identity columns are like this by default. EF does not try to detect this and so if an application does try to specify a value for such a column, then an exception will be thrown by the database.
- IsStoreGeneratedAlways switches off the mechanism whereby EF will try to insert/update values that have been set by the application. However, this flag is only commonly used for concurrency tokens where it is coupled with StoreGenerated.OnAddOrUpdate. In the context of this conversation it can be ignored.